src/main/java/billgates/view/gui/ActionPanel.java

```diff
@@ -0,0 +1,225 @@
+    private final JButton backButton = new ActionButton(" Back", this.backIcon);
+    private final JButton addEntryButton = new ActionButton("Add Entry");
+    private final JButton deleteEntryButton = new ActionButton("Delete Entry");
+    private final JTextArea statisticsTextArea = new ActionTextArea("Statistics");
+
+    public ActionPanel() {
+        this.setLayout(this.layout);
+        this.initSignInPanel();
+        this.initButtonTextArea();
+        this.initBorder();
+        this.setPreferredSize(new Dimension(DEFAULT_WIDTH, DEFAULT_HEIGHT));
+    }
+
+    private void initSignInPanel() {
+        this.add(this.signInPanel);
+        // Set the size of signInPanel
+        this.signInPanel.setMaximumSize(new Dimension(DEFAULT_SIGN_IN_PANEL_WIDTH, DEFAULT_SIGN_IN_PANEL_HEIGHT));
+
+        // add and layout components
+        // username label
+        this.signInPanel.add(this.usernameLabel);
+        this.signInLayout.putConstraint(SpringLayout.WEST, this.usernameLabel, HORIZONTAL_GAP,
+                SpringLayout.WEST, this.signInPanel);
+        this.signInLayout.putConstraint(SpringLayout.NORTH, this.usernameLabel, VERTICAL_GAP,
+                SpringLayout.NORTH, this.signInPanel);
+
+        // username field
+        this.signInPanel.add(this.usernameField);
+        this.signInLayout.putConstraint(SpringLayout.WEST, this.usernameField, HORIZONTAL_GAP,
+                SpringLayout.EAST, this.usernameLabel);
+        this.signInLayout.putConstraint(SpringLayout.NORTH, this.usernameField, VERTICAL_GAP,
+                SpringLayout.NORTH, this.signInPanel);
+
+        // password label
+        this.signInPanel.add(this.passwordLabel);
+        this.signInLayout.putConstraint(SpringLayout.WEST, this.passwordLabel, HORIZONTAL_GAP,
+                SpringLayout.WEST, this.signInPanel);
+        this.signInLayout.putConstraint(SpringLayout.NORTH, this.passwordLabel, VERTICAL_GAP,
+                SpringLayout.SOUTH, this.usernameLabel);
+
+        // password field
+        this.signInPanel.add(this.passwordField);
+        this.signInLayout.putConstraint(SpringLayout.WEST, this.passwordField, 0,
+                SpringLayout.WEST, this.usernameField);
+        this.signInLayout.putConstraint(SpringLayout.NORTH, this.passwordField, VERTICAL_GAP,
+                SpringLayout.SOUTH, this.usernameField);
+    }
+
+    private void initButtonTextArea() {
+        // Components with gaps
+        // signInButton
+        this.add(this.signInButton);
+        this.signInButton.setAlignmentX(CENTER_ALIGNMENT);
+        this.add(Box.createRigidArea(new Dimension(0,VERTICAL_GAP)));
+        // Sign in event
+        this.signInButton.addActionListener((e -> this.signIn()));
+
+        // signOutButton
+        this.add(this.signOutButton);
+        this.signOutButton.setAlignmentX(CENTER_ALIGNMENT);
+        this.add(Box.createRigidArea(new Dimension(0,VERTICAL_GAP)));
```

Comment:
Maybe @ScottCTD will be mad on convention issues here, because there's no space after a comma? 
There's numerous more "no spaces" before and after this comment :3